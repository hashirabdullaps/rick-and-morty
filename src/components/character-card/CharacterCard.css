.ant-card-cover img {
  transform-origin: 100 100;
  transition: transform 0.5s ease;
  flex-shrink: 0;
  min-height: 100%;
  min-width: 100%;
  max-height: 100%;
}

.ant-card-cover img:hover {
  transform: scale(0.95);
}

.ant-card-body {
  background: #f8efc6;
}
