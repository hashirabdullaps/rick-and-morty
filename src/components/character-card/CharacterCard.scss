.character-card {

  .ant-card {
    transform-origin: 100 100;
    transition: transform 0.5s ease;
    overflow: hidden;
    border-radius: 1em;
    margin: 2em;
    border-width: 2px;
    border-color: white;
  }

  .ant-card:hover {
    transform: scale(1.07);
    border-color: #e45f6f;
  }
  
  .ant-card-cover img {
    transform-origin: 100 100;
    transition: transform 0.5s ease;
    flex-shrink: 0;
    min-height: 100%;
    min-width: 100%;
    max-height: 100%;
    border-top-left-radius: 1em;
    border-top-right-radius: 1em;
  }

  .ant-card-cover img:hover {
    transform: scale(1.05);
    border-top-left-radius: 0.9em;
    border-top-right-radius: 0.9em;
  }

}
