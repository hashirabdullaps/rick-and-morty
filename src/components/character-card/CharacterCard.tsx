import { useNavigate } from "react-router-dom";
import { Card } from "antd";

import ICharacter from '../../types/ICharacter'
//styles
import "./CharacterCard.css";

const { Meta } = Card;

const CharacterCard = ({ id, name, image, status, location }: ICharacter) => {
    const navigate = useNavigate();
    return (
        <Card
            hoverable
            style={{ width: 300 }}
            cover={
                <img className="character_img" src={image} alt={name} />
            }>
            <Meta
                title={name}
                description={location?.name}
            />
            <p><b>Status: </b>{status}</p>
        </Card >
    );
};

export default CharacterCard;
