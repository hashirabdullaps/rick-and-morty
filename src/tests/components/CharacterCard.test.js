import React from 'react';
import renderer from 'react-test-renderer';
import CharacterCard from '../../components/character-card/CharacterCard';
import { BrowserRouter } from "react-router-dom";

describe('CharacterCard component', () => {

    const initialState = {
        id: 1,
        name: "test",
        image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
        location: "earth",
        status: 'Alive'
    };

    test('should test CharacterCard component', () => {
        const component = renderer.create(
            <BrowserRouter>
                <CharacterCard {...initialState} />
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
