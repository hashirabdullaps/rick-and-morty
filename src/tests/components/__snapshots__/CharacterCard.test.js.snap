// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard component should test CharacterCard component 1`] = `
<div
  className="character-card"
>
  <div
    className="ant-card ant-card-bordered ant-card-hoverable"
    onClick={[Function]}
    style={
      Object {
        "width": 300,
      }
    }
  >
    <div
      className="ant-card-cover"
    >
      <img
        alt="test"
        className="character_img"
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-card-meta"
      >
        <div
          className="ant-card-meta-detail"
        >
          <div
            className="ant-card-meta-title"
          >
            test
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
